### DISCUSSION TITLE
Porn flips

### DISCUSSION THREAD

**Discussion by KruciferX** [idena-network/idena-desktop#633] [Ideas]
Recently I've noticed some flips being made with hardcore porn images. I don't want my kids looking at those while solving flips.

Reporting those images is useless because those flips are already shown (and some flips like that don't even get reported which is puzzling and worrying me).

But there are ways to discard such images during flip creation/submission directly from client.

**Comment by midenaio** [comment]
To be mitigated by using a nude detection filter both for the flip submission and validation session.

**Comment by KruciferX** [comment]
I did some searching and here are all the repos I could find on github that can possibly be used.

This one is already ported to golang:
https://github.com/koyachi/go-nude

If that's not good, then it's a lot of work to find a suitable one from the list below, but that's beyond the scope of my knowledge. :)

Maybe someone from the community with coding skills can help narrowing it down.

https://github.com/pa7/nude.js
https://github.com/FreebieStock/php-nudity-detector
https://github.com/bedapudi6788/NudeNet
https://github.com/sajithm/nsfw-v1
https://github.com/sajithm/nsfw-v2
https://github.com/chebyrash/censor
https://github.com/infinitered/nsfwjs
https://github.com/MaybeShewill-CV/nsfw-classification-tensorflow
https://github.com/usama093/tensorflow-open_nsfw
https://github.com/usama093/Keras_open_nsfw
https://github.com/EugenCepoi/nsfw_api
https://github.com/emiliantolo/pytorch_nsfw_model
https://github.com/thiventura/OpenModelsAppliedToForensicImages
https://github.com/deepanshu-yadav/NSFW-Classifier
https://github.com/lakshaychhabra/NSFW-Detection-DL
https://github.com/vqmalic/se_seg
https://github.com/yahoo/open_nsfw
https://github.com/ph1ps/Nudity-CoreML

**Comment by jugi1** [comment]
I still think  some words should be removed from the list.
No need for ORGY and RAPE, and similar

**Comment by KruciferX** [comment]
> I still think some words should be removed from the list.
> No need for ORGY and RAPE, and similar

I agree with that.

There's no good story to be made with those words.

**Comment by jugi1** [comment]
Still think word Orgy is ok?
Every time result is porn flips:
https://i.imgur.com/AHO3kru.jpg

**Comment by jugi1** [comment]
https://i.imgur.com/g4x8B25.png
https://scan.idena.io/flip/bafkreib3tvtckbbpnpicbouuympzujggedn6qevgmspnpwiaebxqislz3y

**Comment by gabrieldonadel** [comment]
> Recently I've noticed some flips being made with hardcore porn images. I don't want my kids looking at those while solving flips.
> 
> Reporting those images is useless because those flips are already shown (and some flips like that don't even get reported which is puzzling and worrying me).
> 
> But there are ways to discard such images during flip creation/submission directly from client.

One of my concerns about making this detection only on the client-side is that a user could run an adulterated version of the app that simply skips the verification at all. But, I also believe that at least having the filter on the client-side would stop the majority of these cases.
